ContactsManagerBackend/ContactsManager/ContactsManagerDAL/ContactsProvider.cs

```diff
@@ -223,6 +239,50 @@ public Contact UpdateContact(Guid Id, Contact contact)
                     }
                 }
 
+                if (contact.Phones != null && contact.Phones.Count > 0)
+                {
+                    var phonesToCreate = contact.Phones.Where(p => p.IsNew).ToList();
```

Comment:
You should compare data for update with existing data to determine what to create, update and delete.